### ISSUE TITLE
Implement rhboot UEFI SHIM loader

### ISSUE THREAD

**Issue by lukehinds** [keylime/python-keylime#63]
# Environment
 * OS / version: RHEL and family
 * Processor architecture: N/A
 * TPM Manufacturer: N/A
 * Keylime version: master

# Description 

This is an initial issue to introduce the work of implementing the rhboot TPM compliant UEFI shim loader [0] .

The issue should be followed up with a spec outlining approach, considerations and core tasks.  

[0] https://github.com/rhboot/shim

**Comment by lukehinds** [comment]
Some more details after talking with the rhboot devs.

> What is the current status of the rhboot shim? Is this included into
> a certain fedora release?


Yes, shim is included in both Fedora and RHEL. It's installed by default
if you do an EFI install.

> In the readme it states "If you're using the grub2 TPM patchset we
> cary in Fedora, the kernel, initramfs, and any multiboot modules
> loaded are measured into PCR9." 

We are also carrying the TPM2 patches in the Fedora/RHEL grub2 packages.

> In keylime we had it measuring TrustedGrub2 on PCR-10 [0], and would like
> to see if I can get this working with rhboot instead
> 
> [0]
> https://github.com/Rohde-Schwarz-Cybersecurity/TrustedGRUB2#13-measurements-in-short

This should effectively do all of that, with the exception that the TPM
2 Client TIS has the firmware extend the bootloader hash into PCR4,
and we use PCR9 as you had noticed for the kernel and initramfs.  We're
not the only vendor to have moved them to separate PCRs, either, though
we've done it differently (but compatibly) with MSFT's implementation.
Basically, our PCR4 usage is from here:
https://docs.microsoft.com/en-us/previous-versions/windows/hardware/hck/jj923068(v=vs.85)#appendix_a__static_root_of_trust_measurements
And we've decided to also use PCR9 for kernel+initrd to make it possible
to attest a value specific to them.

**Comment by lukehinds** [comment]
Quick update, this is working now..I will put up a blog post. Will close this for now
